bundles/io.openliberty.tools.eclipse.ui/src/io/openliberty/tools/eclipse/DebugModeHandler.java

```diff
@@ -0,0 +1,462 @@
+                            "Unable to attach debugger to host: " + DEFAULT_ATTACH_HOST + " and port: " + debugPort, e);
+                }
+
+                return Status.OK_STATUS;
+            }
+        };
+
+        // Register a listener with the terminal tab controller. This listener handles cleanup when the terminal or terminal tab is
+        // terminated
+        // while actively processing work.
+        TerminalListener terminalListener = new TerminalListener() {
+            /**
+             * {@inheritDoc}
+             */
+            @Override
+            public void cleanup() {
+                job.cancel();
+            }
+        };
+        devModeOps.registerTerminalListener(project.getIProject().getName(), terminalListener);
+
+        // Register a job change listener. This listener performs job completion processing.
+        job.addJobChangeListener(new JobChangeAdapter() {
+            @Override
+            public void done(IJobChangeEvent event) {
+                devModeOps.deregisterTerminalListener(projectName, terminalListener);
+                IStatus result = event.getResult();
+                Throwable t = result.getException();
+                if (t != null) {
+                    if (Trace.isEnabled()) {
+                        Trace.getTracer().trace(Trace.TRACE_UI, t.getMessage(), t);
+                    }
+
+                    ErrorHandler.processErrorMessage(t.getMessage(), t, false);
+                }
+            }
+        });
+
+        job.schedule();
+    }
+
+    /**
+     * Returns the default path of the server.env file after Liberty server deployment.
+     * 
+     * @param project The project for which this operations is being performed.
+     * 
+     * @return The default path of the server.env file after Liberty server deployment.
+     * 
+     * @throws Exception
+     */
+    private Path getServerEnvPath(Project project) throws Exception {
+        String projectPath = project.getPath();
+        Path basePath = null;
+        BuildType buildType = project.getBuildType();
+        if (buildType == Project.BuildType.MAVEN) {
+            basePath = Paths.get(projectPath, "target", "liberty", "wlp", "usr", "servers");
+        } else if (buildType == Project.BuildType.GRADLE) {
+            basePath = Paths.get(projectPath, "build", "wlp", "usr", "servers");
+        } else {
+            throw new Exception("Unexpected project build type: " + buildType + ". Project" + project.getIProject().getName()
+                    + "does not appear to be a Maven or Gradle built project.");
+        }
+
+        // Make sure the base path exists. If not return null.
+        File basePathFile = new File(basePath.toString());
+        if (!basePathFile.exists()) {
+            return null;
+        }
+
+        // Search for a server.env and return it's path.
+        Path filePattern = Paths.get(basePath.toString(), "*", "server.env");
```

Comment:
This doesn't work for me... I tried Semeru & Temurin JDKs .   The related tests fail locally so I'm not sure how it passes in GHA.  I wonder what JDK distro it's using there?  

For me it results in:
java.nio.file.InvalidPathException: Illegal char <*> at index 119: C:\wdt2\liberty-tools-eclipse\tests\resources\applications\maven\liberty-maven-test-app\target\liberty\wlp\usr\servers\*\server.env
        at java.base/sun.nio.fs.WindowsPathParser.normalize(WindowsPathParser.java:182)
        at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:153)
        at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:77)
        at java.base/sun.nio.fs.WindowsPath.parse(WindowsPath.java:92)
        at java.base/sun.nio.fs.WindowsFileSystem.getPath(WindowsFileSystem.java:232)
        at java.base/java.nio.file.Path.of(Path.java:147)
        at java.base/java.nio.file.Paths.get(Paths.java:69)
        at io.openliberty.tools.eclipse.DebugModeHandler.getServerEnvPath(DebugModeHandler.java:263)

I guess look at something like: ?

https://stackoverflow.com/a/30089290/1499766

Reply:
Looks good now